x-pack/plugins/reporting/server/deprecations/reporting_role.ts

```diff
@@ -45,101 +42,7 @@ export async function getDeprecationsInfo(
-        ` Set "xpack.reporting.roles.enabled" to "false" to adopt the future behavior before upgrading.`,
-    }),
-    manualSteps: (usersRoles: string) => [
-      ...(usingDeprecatedConfig
-        ? [
-            i18n.translate('xpack.reporting.deprecations.reportingRoleUsers.manualStepOne', {
-              defaultMessage: `Set "xpack.reporting.roles.enabled" to "false" in kibana.yml.`,
-            }),
-            i18n.translate('xpack.reporting.deprecations.reportingRoleUsers.manualStepTwo', {
-              defaultMessage: `Remove "xpack.reporting.roles.allow" in kibana.yml, if present.`,
-            }),
-          ]
-        : []),
-
-      i18n.translate('xpack.reporting.deprecations.reportingRoleUsers.manualStepThree', {
-        defaultMessage:
-          `Go to Management > Security > Roles to create one or more roles that grant` +
-          ` the Kibana application privilege for Reporting.`,
-      }),
-      i18n.translate('xpack.reporting.deprecations.reportingRoleUsers.manualStepFour', {
-        defaultMessage: `Grant Reporting privileges to users by assigning one of the new roles.`,
-      }),
-      i18n.translate('xpack.reporting.deprecations.reportingRoleUsers.manualStepFive', {
-        defaultMessage:
-          `Remove the "reporting_user" role from all users and add the custom role.` +
-          ` The affected users are: {usersRoles}.`,
-        values: { usersRoles },
-      }),
-    ],
-  };
-
-  let users: SecurityGetUserResponse;
-  try {
-    users = await client.security.getUser();
-  } catch (err) {
-    const { logger } = reportingCore.getPluginSetupDeps();
-    if (deprecations.getErrorStatusCode(err) === 403) {
-      logger.warn(
-        `Failed to retrieve users when checking for deprecations:` +
-          ` the "manage_security" cluster privilege is required.`
-      );
-    } else {
-      logger.error(
-        `Failed to retrieve users when checking for deprecations,` +
-          ` unexpected error: ${deprecations.getDetailedErrorMessage(err)}.`
-      );
-    }
-    return deprecations.deprecationError(strings.title, err, docLinks);
-  }
-
-  const reportingUsers = Object.entries(users).reduce((userSet, current) => {
-    const [userName, user] = current;
-    const foundRole = user.roles.find((role) => deprecatedRoles.includes(role));
-    if (foundRole) {
-      userSet.push(`${userName}[${foundRole}]`);
-    }
-    return userSet;
-  }, [] as string[]);
-
-  if (reportingUsers.length === 0) {
-    return [];
-  }
-
-  return [
-    {
-      title: strings.title,
-      message: strings.message,
-      correctiveActions: { manualSteps: strings.manualSteps(reportingUsers.join(', ')) },
-      level: 'warning',
-      deprecationType: 'feature',
-      documentationUrl: getDocumentationUrl(reportingCore.getKibanaPackageInfo().branch),
-    },
-  ];
+  return [...(await getRoleMappingsDeprecations(client, reportingCore, deprecatedRoles, docLinks))];
```

Comment:
This can be simplified:
```suggestion
  return await getRoleMappingsDeprecations(client, reportingCore, deprecatedRoles, docLinks);
```